server/webapp/WEB-INF/rails/webpack/models/config_repos/types.ts

```diff
@@ -142,6 +146,23 @@ export class ConfigRepo implements ValidatableMixin {
     }
     return configurations;
   }
+
+  matches(textToMatch: string): boolean {
+    if (!textToMatch) {
+      return true;
+    }
+
+    const id             = this.id();
+    const goodRevision   = this.lastParse() && this.lastParse()!.goodModification && this.lastParse()!.goodModification!.revision;
+    const latestRevision = this.lastParse() && this.lastParse()!.latestParsedModification && this.lastParse()!.latestParsedModification!.revision;
```

Comment:
The type for lastParse is defined as `Stream<ParseInfo | null>`. So using the `!` operator here can be potentially unsafe and lead to runtime errors.

Even if you're sure that the value won't be null when this method is called, it can lead to bugs in the future.